robottelo/common/ssh.py

```diff
@@ -94,32 +94,27 @@ def _get_connection(
         logger.info('Destroyed Paramiko client {0}'.format(client_id))
 
 
-def upload_file(local_file, remote_file=None):
-    """
-    Uploads a remote file to a normal server or
-    uploads a file to sauce labs via VPN tunnel.
-    """
-
-    remote = int(conf.properties['main.remote'])
+def upload_file(local_file, remote_file=None, hostname=None):
+    """Upload a local file to a remote machine. If ``hostname`` is not provided
+    will be used the server.hostname from the cofiguration.
 
+    """
     if not remote_file:
         remote_file = local_file
-
-    if not remote:
-        with _get_connection() as connection:
-            sftp = connection.open_sftp()
-            sftp.put(local_file, remote_file)
-            sftp.close()
-    # TODO: Upload file to sauce labs via VPN tunnel in the else part.
+    with _get_connection(hostname=hostname) as connection:
+        sftp = connection.open_sftp()
```

Comment:
Is there a context manager for this too? What if `sftp.put` fails? Will `sftp.close()` be called anyway?